chapter/9/streaming.md

```diff
@@ -1,10 +1,162 @@
+
+## How to process data stream
+
+Now we know what the stream looks like and how do we ensure that the data in the stream are successfully processed. We would then talk about the processors that cosume the data stream. There are two main approaches in processing data stream. The first approach is the continuous queries model, similar to TelegraphCQ, where the queries keep running and the arrival of data intiates the processing. Another approach is micro-batching, where the streaming computation becomes a series of stateless, deterministic batch computations on batch of stream, where certain timer would triger the processing on the batch in those systems. We would discuss Apach Storm as an example for the fist design and Spark Streaming, Naiad and Google Dataflow are examples of the second approach. These systems not only differ in the way how they process stream, but also how they ensure fault-tolerance which is one of the most important aspects of large scale distributed system.
+
+### a) Continuous queries (operators) on each tuple 
+
+- Apache Storm
+
+After MapReduce, Hadoop, and the related batch processing system came out, the data can be processed at scales previously unthinkable. However, as we stated before, a realtime large scale data processing becomes more and more important for many businesses. *Apache Storm* {% cite apachestorm --file streaming %} is actually one of the first system that can be discribed as "Hadoop of realtime" that feed the needs. Users can process messages in a way that doesn't lose data and also scalable with the primitives provided by *Storm*.
```

Comment:
I think that *technically*, Storm isn't a real-time system. It's just a stream processing system, source: https://www.quora.com/Whats-the-difference-between-real-time-processing-and-stream-processing